account_commission_listing_fee/models/commission_settlement.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CommissionSettlement(models.Model):
+    _inherit = "commission.settlement"
+
+    event_ids = fields.Many2many(
+        "event.event", compute="_compute_event_ids", string="Events", store=True
+    )
+    number_of_items = fields.Integer(compute="_compute_number_of_items", store=True)
+
+    @api.depends("line_ids")
+    def _compute_event_ids(self):
+        for settlement in self:
+            events = settlement.line_ids.product_id.event_ids
+            settlement.event_ids = events
+
+    @api.depends("event_ids", "agent_id")
+    def _compute_number_of_items(self):
+        for settlement in self:
+            number_of_product = 0
+            products = self.env["product.product"].search(
+                [("seller_id", "=", settlement.agent_id.id)]
+            )
+            for product in products:
+                if product.event_ids:
+                    check_event = all(
+                        event in settlement.event_ids.ids
+                        for event in product.event_ids.ids
+                    )
+                    if check_event:
+                        number_of_product += 1
+            settlement.number_of_items = number_of_product
```

Comment:
```suggestion
            event_ids = settlement.event_ids.ids
            products = self.env["product.product"].search(
                [("seller_id", "=", settlement.agent_id.id)]
            ).filtered(lambda x: set(x.event_ids.ids).issubset(event_ids))
            settlement.number_of_items = len(products)
```
Would this work? Haven't tested myself.